File: week1-array-string/problem01-배열두배만들기/sinhye.java
// 문제: 배열 두 배 만들기
// 접근 방식: 새 배열을 만들어 각 요소를 순회하며 2배 값을 저장
// 시간복잡도: O(n)
class Solution {
    public int[] solution(int[] numbers) {
        int[] answer = new int[numbers.length];
        
        for (int i = 0;i < numbers.length;i++) {
            answer[i] = numbers[i] * 2;
        }
        
        return answer;
    }
}

File: week3-hash-sort/problem01-완주하지못한선수/sinhye.cpp
// 문제: 완주하지 못한 선수
// 접근 방식: 해시맵을 이용해서 참가자에서 완주자 제외
// 시간복잡도: O(N)
#include <string>
#include <vector>
#include <unordered_map>
#include <iostream>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> marathon;
    
    for (const string& p : participant) {
        marathon[p]++;
    }
    
    for (const string& c : completion) {
        marathon[c]--;
    }
    
    for (const pair<string, int>& m : marathon) {
        if (m.second > 0) {
            answer = m.first;
        }
    }
    
    return answer;
}
